addons/purchase/models/purchase.py

```diff
@@ -162,6 +150,11 @@ def create(self, vals):
             vals['name'] = self.env['ir.sequence'].next_by_code('purchase.order') or '/'
         return super(PurchaseOrder, self).create(vals)
 
+    def write(self, vals):
+        if vals.get('date_planned'):
+            self.order_line.filtered(lambda x: x.state not in ('purchase', 'done', 'cancle')).write({'date_planned': vals['date_planned']})
```

Comment:
`cancle`???

If use `self`, so must be use `mapped ` on `order_line`, can raise `singleton error`

Reply:
suggestion of @sle-odoo  is better, not filter by state and use `mapped`, solved my comments